airbyte-integrations/connectors/source-twilio/source_twilio/streams.py

```diff
@@ -107,28 +115,33 @@ def incremental_filter_field(self) -> str:
         return: date filter query parameter name
         """
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        """
-        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
-        and returning an updated state object.
-        """
-        latest_benchmark = pendulum.parse(latest_record[self.cursor_field], strict=False).strftime(self.time_filter_template)
-        if current_stream_state.get(self.cursor_field):
-            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
-        return {self.cursor_field: latest_benchmark}
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if self._cursor_value:
+            return {self.cursor_field: self._cursor_value}
+        else:
+            return {self.cursor_field: self._start_date}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = max(value[self.cursor_field], self._cursor_value)
 
     def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
         params = super().request_params(stream_state=stream_state, **kwargs)
-        start_date = stream_state.get(self.cursor_field) or self._start_date
+        start_date = stream_state[self.cursor_field] if stream_state.get(self.cursor_field) else self.start_date
         if start_date:
-            params.update({self.incremental_filter_field: pendulum.parse(start_date, strict=False).strftime(self.time_filter_template)})
+            params.update(
+                {
+                    self.incremental_filter_field: (
+                        pendulum.parse(start_date, strict=False) - timedelta(minutes=self._lookback_window)
```

Comment:
nit to avoid timedelta import
```suggestion
                        pendulum.parse(start_date, strict=False) - pendulum.duration(minutes=self._lookback_window)
```